src/features/customerList/customerListTableFilters/CustomerListTableFiltersForm.tsx

```diff
@@ -0,0 +1,426 @@
+  winterStorageGridAreaOptions?: Option[];
+  winterStoragePlaceOptions?: Option[];
+  winterStorageAreaOptions?: Option[];
+
+  filters: ControlledFilters;
+
+  onFieldChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onSubmit: (e: React.FormEvent<HTMLFormElement>, filters: ControlledFilters) => void;
+  onReset: (...fieldsToReset: (keyof ControlledFilters)[]) => void;
+}
+
+const CustomerListTableFiltersForm = ({
+  customerGroupOptions,
+  boatTypeOptions,
+  leaseStatusOptions,
+  harborOptions,
+  pierOptions,
+  berthOptions,
+  winterStoragePlaceOptions,
+  winterStorageAreaOptions,
+  winterStorageGridAreaOptions,
+  filters: {
+    customerGroups,
+    boatTypeIds,
+    leaseStatuses,
+    harborIds,
+    pierId,
+    berthId,
+    winterStorageGridAreaIds,
+    winterStoragePlaceId,
+    winterStorageAreaIds,
+    startDate,
+    endDate,
+    moreThanOneBerthOrWinterStorage,
+  },
+  onFieldChange,
+  onSubmit,
+  onReset,
+}: Props) => {
+  const { t } = useTranslation();
+
+  const [isSectionEnabled, setSectionEnabled] = React.useState(() => ({
+    berthSection: harborIds?.length > 0,
+    winterStorageSection: winterStorageGridAreaIds?.length > 0,
+    unmarkedWinterStorageSection: winterStorageAreaIds?.length > 0,
+    dateSection: Boolean(startDate || endDate),
+  }));
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) =>
+    onSubmit(e, {
+      customerGroups,
+      boatTypeIds,
+      leaseStatuses,
+      harborIds,
+      pierId,
+      berthId,
+      winterStorageGridAreaIds,
+      winterStoragePlaceId,
+      winterStorageAreaIds,
+      startDate,
+      endDate,
+      moreThanOneBerthOrWinterStorage,
+    });
+
+  const handleSectionChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const sectionName = e.target.name as keyof Section;
+
+    setSectionEnabled((prev) => {
+      const nextValue = !prev[sectionName];
+
+      if (nextValue === false) {
+        const fieldResetMap: Record<keyof Section, (keyof ControlledFilters)[]> = {
+          berthSection: ['harborIds', 'pierId', 'berthId'],
+          winterStorageSection: ['winterStorageGridAreaIds', 'winterStoragePlaceId'],
+          unmarkedWinterStorageSection: ['winterStorageAreaIds'],
+          dateSection: ['startDate', 'endDate'],
+        };
+
+        onReset(...fieldResetMap[sectionName]);
+      }
+
+      return {
+        ...prev,
+        [sectionName]: !prev[sectionName],
+      };
+    });
+  };
+
+  const handleReset = () => {
+    setSectionEnabled({
+      berthSection: false,
+      winterStorageSection: false,
+      unmarkedWinterStorageSection: false,
+      dateSection: false,
+    });
+    onReset(
+      'customerGroups',
+      'boatTypeIds',
+      'leaseStatuses',
+      'harborIds',
+      'pierId',
+      'berthId',
+      'winterStorageGridAreaIds',
+      'winterStoragePlaceId',
+      'winterStorageAreaIds',
+      'startDate',
+      'endDate',
+      'moreThanOneBerthOrWinterStorage'
+    );
```

Comment:
Maybe something like `onResetAll` would make it clearer that this resets the whole form.